app/queries/servers/channel.ts

```diff
@@ -431,3 +432,99 @@ export function queryMyChannelsByUnread(database: Database, isUnread: boolean, s
         Q.take(take),
     );
 }
+
+export const observeDirectChannelsByTerm = (database: Database, term: string, take = 20, matchStart = false) => {
+    const onlyDMs = term.startsWith('@') ? "and c.type='D'" : '';
+    const value = Q.sanitizeLikeString(term.startsWith('@') ? term.substring(1) : term);
+    let username = `u.username like '${value}%'`;
+    let displayname = `c.display_name like '${value}%'`;
+    if (!matchStart) {
+        username = `u.username like '%${value}%' and u.username not like '${value}%'`;
+        displayname = `(c.display_name like '%${value}%' and c.display_name not like '${value}%')`;
+    }
+    const currentUserId = observeCurrentUserId(database);
+    return currentUserId.pipe(
+        switchMap((uId) => {
+            return database.get<MyChannelModel>(CHANNEL).query(
+                Q.unsafeSqlQuery(`select distinct my.* from ${MY_CHANNEL} my
```

Comment:
Why use unsafeSqlQuery? The `Q` functions don't work in this scenario?

Reply:
Q definitely won't allow us to do what the query is doing in this scenario I'm afraid, specially cause of the 
**DISTINCT**, **NOT LIKE** operators